pyimzml/ImzMLWriter.py

```diff
@@ -123,43 +133,69 @@ def __init__(self, output_filename, mz_dtype=np.float64, intensity_dtype=np.floa
         self.imzml_template = self.wheezy_engine.get_template('imzml')
 
         self.spectra = []
+        self.hashes = {} #set this to None to force processed mode.
```

Comment:
OK, added. I didn't want to use lrucache since it would require reading from the file unnecessarily, so I basically made my own so I could control the cache. Also it save the user an install, since cachetools does not come with python2.